### PR TITLE
chore(text-field): remove dark theme

### PR DESCRIPTION
**Pull request by moog16**

### CODE REVIEW

**Comment by aaronhudon** [comment]
Does this imply dark theme support is being pulled from controls, or is this just for the demo pages showing both versions at the same time...?

**Comment by moog16** [comment]
@aaronhudon we are pulling dark theme styles out from scss. Its become too much to manage. We want the end developer to be using our sass mixins to style the components.

**Comment by aaronhudon** [comment]
@moog16 so this means the `.mdc-theme--dark` modifier is also being removed?

**Comment by acdvorak** [comment]
Thanks for asking, @aaronhudon - we always appreciate feedback from valued members of the community!

I know it's disappointing 🙁   But sadly it's necessary to keep our code healthy and maintainable, as @moog16 alluded to.

For a more detailed explanation of our motivation and reasoning, see [my response to #1332](https://github.com/material-components/material-components-web/issues/1332#issuecomment-335724659).

The TL;DR is that instead of using `--theme-dark` classes, you can customize the theme variables (`$mdc-theme-background`) and use our component theme mixins to achieve the same result.

Our goal is to make that relatively easy, so please let us know if it's not. (Keep in mind that we're still actively working on theming and customization, so it's very much a work in progress.)

**Comment by aaronhudon** [comment]
@acdvorak thanks for the followup, I trust a guide on this new method of implementing a dark theme will be posted in the docs?

**Comment by acdvorak** [comment]
@aaronhudon There's an example on our [theme demo page](https://material-components-web.appspot.com/theme/index.html?theme=dark) (the source is in
[`demos/theme/theme-dark.scss`](https://github.com/material-components/material-components-web/blob/f399384be2c607c5618fe181f8b5b049f46e072a/demos/theme/theme-dark.scss)).

In short, you need to A) override `$mdc-theme-background` to a dark color (`#333` or whatever), and B) call a few component theme mixins. We're working on removing the need to call theme mixins, but for now that's the easiest way to do it.

File: packages/mdc-textfield/_variables.scss [hunk 1]
```diff
@@ -15,41 +15,27 @@
 //
 
 $mdc-text-field-error-on-light: #d50000;
```
Comment (acdvorak) [inline packages/mdc-textfield/_variables.scss]: We can probably remove the `-on-error` suffix from all of the remaining color variables

Comment (moog16) [reply packages/mdc-textfield/_variables.scss]: I removed all the -on-light as well as -light-* variable names.

File: packages/mdc-textfield/_variables.scss [hunk 2]
```diff
@@ -14,15 +14,14 @@
...
-$mdc-text-field-underline-on-light-idle: rgba(black, .5);
-$mdc-text-field-light-background: rgba(white, 1);
-$mdc-text-field-light-label: rgba(black, .6);
-$mdc-text-field-light-placeholder: rgba(black, .38);
+$mdc-text-field-error: #d50000;
+$mdc-text-field-divider: rgba(black, .12);
+$mdc-text-field-disabled-border: rgba(#231f20, .26);
+$mdc-text-field-disabled-icon: rgba(black, .3);
+$mdc-text-field-underline: rgba(black, 1);
+$mdc-text-field-underline: rgba(black, .5);
```
Comment (acdvorak) [inline packages/mdc-textfield/_variables.scss]: Add `-idle` back

File: packages/mdc-textfield/mdc-text-field.scss [hunk 3]
```diff
@@ -33,14 +33,14 @@
 // postcss-bem-linter: define text-field
 
 .mdc-text-field {
-  @include mdc-text-field-bottom-line-color($mdc-text-field-underline-on-light-idle);
-  @include mdc-text-field-hover-bottom-line-color($mdc-text-field-underline-on-light);
+  @include mdc-text-field-bottom-line-color($mdc-text-field-underline);
```
Comment (acdvorak) [inline packages/mdc-textfield/mdc-text-field.scss]: Add `-idle` back

File: packages/mdc-textfield/_variables.scss [hunk 4]
```diff
@@ -14,15 +14,14 @@
...
-$mdc-text-field-underline-on-light: rgba(black, 1);
-$mdc-text-field-underline-on-light-idle: rgba(black, .5);
-$mdc-text-field-light-background: rgba(white, 1);
-$mdc-text-field-light-label: rgba(black, .6);
-$mdc-text-field-light-placeholder: rgba(black, .38);
+$mdc-text-field-error: #d50000;
+$mdc-text-field-divider: rgba(black, .12);
+$mdc-text-field-disabled-border: rgba(#231f20, .26);
+$mdc-text-field-disabled-icon: rgba(black, .3);
+$mdc-text-field-underline: rgba(black, 1);
```
Comment (acdvorak) [inline packages/mdc-textfield/_variables.scss]: Can we rename this to `$mdc-text-field-underline-hover`, since that's what it's used for?

File: packages/mdc-textfield/mdc-text-field.scss [hunk 5]
```diff
@@ -33,14 +33,14 @@
...
+  @include mdc-text-field-hover-bottom-line-color($mdc-text-field-underline);
```
Comment (acdvorak) [inline packages/mdc-textfield/mdc-text-field.scss]: `$mdc-text-field-underline-hover`

File: packages/mdc-textfield/mdc-text-field.scss [hunk 6]
```diff
@@ -33,14 +33,14 @@
...
   @include mdc-text-field-focused-bottom-line-color(primary);
   @include mdc-text-field-ink-color(text-primary-on-light);
-  @include mdc-text-field-label-color($mdc-text-field-light-label);
+  @include mdc-text-field-label-color($mdc-text-field-label);
   @include mdc-text-field-helper-text-color(text-hint-on-light);
-  @include mdc-text-field-fullwidth-bottom-line-color($mdc-text-field-divider-on-light);
-  @include mdc-text-field-icon-color($mdc-text-field-underline-on-light);
+  @include mdc-text-field-fullwidth-bottom-line-color($mdc-text-field-divider);
+  @include mdc-text-field-icon-color($mdc-text-field-underline);
```
Comment (acdvorak) [inline packages/mdc-textfield/mdc-text-field.scss]: `$mdc-text-field-underline-hover`

File: demos/text-field.html [hunk 7]
```diff
@@ -231,11 +230,21 @@ <h2>Text Field box</h2>
...
+          <input id="box-persistent-helper-text" type="checkbox" disabled>
+          <label for="box-persistent-helper-text">Make helper text persistent</label>
+        </div>
         <script>
           demoReady(function() {
+            var section = document.getElementById('text-field-box-example');
             var tfEl = document.getElementById('tf-box-example');
             var tf = new mdc.textField.MDCTextField(tfEl);
             var wrapper = document.getElementById('demo-tf-box-wrapper');
+            var helperText = section.querySelector('.mdc-text-field-helper-text');
```
Comment (acdvorak) [inline demos/text-field.html]: Let's give this element an `id` attribute and use `getElementById` instead

File: demos/text-field.html [hunk 8]
```diff
@@ -231,11 +230,21 @@ <h2>Text Field box</h2>
           <input id="box-required" type="checkbox">
           <label for="box-required">Required</label>
```
Comment (acdvorak) [inline demos/text-field.html]: While we're here, can we rename this label to "Required and must be at least 8 characters"?

File: demos/text-field.html (hunk 7 above)
Comment (moog16) [reply demos/text-field.html]: Also will remove line 243...no longer using section

**Comment by theenoahmason** [comment]
@moog The dark theme is being removed from the entire package? as in even a global `.mdc-theme--dark` will not work?

If yeah, that's kind of a bummer! Will any of the CSS vars be removed like `...on-dark`, `...on-light`?

Will classes be removed like `...on-dark`, `...on-light`?

**Comment by moog16** [comment]
@theenoahmason yes `.mdc-theme--dark` and `@mixin mdc-theme-dark` will be removed from the package. Having all the dark theme support was complicating our CSS, and as a team we decided to remove it. We have been moving towards sass mixins to achieve the same affect. 

We will not be removing CSS vars as you mentioned (`*-on-dark`, `*-on-light`, etc). These will remain as they do not require extra css rules to support them.
